lib/rules/indent.js

```diff
@@ -741,6 +750,24 @@ module.exports = {
                 checkIndentInArrayOrObjectBlock(node);
             },
 
+            MemberExpression: function(node) {
+                if (isSingleLineNode(node)) {
+                    return;
+                }
+
+                var propertyIndent = getNodeIndent(node) + indentSize * options.MemberExpression;
+
+                var checkNodes = [node.property];
+
+                var dot = context.getTokenBefore(node.property);
+
+                if (dot.type === "Punctuator" && dot.value === ".") {
```

Comment:
This logic here is a bit confusing to me. Why are we checking the dot additionally? Here's what I think:

- If the dot is on the previous line, we only need to check the `MemberExpression` `node`.
- If the dot is on the same line with `MemberExpression`, then we only need to check the dot's indentation.

Am I right?